driver-core/src/main/com/mongodb/internal/connection/ConcurrentPool.java

```diff
@@ -143,7 +145,8 @@ public T get(final long timeout, final TimeUnit timeUnit) {
 
         T t = available.pollLast();
         if (t == null) {
-            t = createNewAndReleasePermitIfFailure(false);
+            t = createNewAndReleasePermitIfFailure(null);
+            assert t != null : "new item must not be null if postCreate is null";
```

Comment:
Discussed this with @rozza and @jyemin. Decided to add `assert...` methods throwing `AssertionError` to `com.mongodb.assertions.Assertions` and used them instead of the Java `assert` statement.

Done.